index.js

```diff
@@ -6,8 +6,29 @@ const replace = require('broccoli-replace')
 
 module.exports = {
   name: 'seamless-immutable',
+  /**
+     This method climbs up the hierarchy of addons
+     up to the host application.
+     This prevents previous addons (prior to `this.import`, ca 2.7.0)
+     to break at importing assets when they are used nested in other addons.
+     @private
+     @method _findHost
+   */
+  _findHost: function() {
+    var current = this;
+    var app;
+
+    // Keep iterating upward until we don't have a grandparent.
+    // Has to do this grandparent check because at some point we hit the project.
+    do {
+      app = current.app || app;
+    } while (current.parent.parent && (current = current.parent));
```

Comment:
Is it ever possible for `current.parent` to not be defined? Might be better to do:
```js
do {
  app = current.app || app;
} while (current && current.parent && current.parent.parent && (current = current.parent));
```